src/main/java/org/spongepowered/common/data/processor/data/entity/FlyingDataProcessor.java

```diff
@@ -0,0 +1,154 @@
+    }
+
+    @SuppressWarnings("unused")
+    @Override
+    public Optional<FlyingData> from(DataHolder dataHolder) {
+        if (supports(dataHolder)) {
+            boolean flying = ((net.minecraft.entity.Entity) dataHolder).isAirBorne;
+            if (dataHolder instanceof EntityPlayer) {
+                EntityPlayer player = (EntityPlayer) dataHolder;
+                flying = player.capabilities.isFlying;
+            }
+            final SpongeFlyingData flyingData = new SpongeFlyingData(flying);
+            return Optional.<FlyingData>of(flyingData);
+        } else {
+            return Optional.absent();
+        }
+
+    }
+
+    @Override
+    public Optional<FlyingData> fill(DataHolder dataHolder, FlyingData manipulator, MergeFunction overlap) {
+        if (supports(dataHolder)) {
+            final FlyingData merged =
+                    checkNotNull(overlap, "MergeFunction cannot be null!").merge(checkNotNull(manipulator).copy(), from(dataHolder).orNull());
+            manipulator.set(Keys.IS_FLYING, merged.flying().get());
+            return Optional.of(manipulator);
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<FlyingData> fill(DataContainer container, FlyingData flyingData) {
+        flyingData.set(Keys.IS_FLYING, getData(container, Keys.IS_FLYING));
+        return Optional.of(flyingData);
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, FlyingData manipulator, MergeFunction function) {
+        if (this.supports(dataHolder)) {
+            final FlyingData merged =
+                    checkNotNull(function, "MergeFunction cannot be null!").merge(checkNotNull(manipulator).copy(), from(dataHolder).orNull());
+            manipulator.set(Keys.IS_FLYING, merged.flying().get());
+            final boolean value = merged.flying().get();
+            if (dataHolder instanceof EntityPlayer) {
+                EntityPlayer player = (EntityPlayer) dataHolder;
+                player.capabilities.isFlying = value;
+            } else {
+                net.minecraft.entity.Entity entity = (net.minecraft.entity.Entity) dataHolder;
+                entity.isAirBorne = value;
+            }
+            return DataTransactionBuilder.successReplaceResult(new ImmutableSpongeValue<Boolean>(Keys.IS_FLYING, !(value)),
+                    new ImmutableSpongeValue<Boolean>(Keys.IS_FLYING, value));
+        }
+        return DataTransactionBuilder.failResult(manipulator.getValues());
+    }
+
+    @Override
+    public Optional<ImmutableFlyingData> with(Key<? extends BaseValue<?>> key, Object value, ImmutableFlyingData immutable) {
+        if (key.equals(Keys.IS_FLYING)) {
+            return Optional.<ImmutableFlyingData>of(ImmutableDataCachingUtil.getManipulator(ImmutableSpongeFlyingData.class, value));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public DataTransactionResult remove(DataHolder dataHolder) {
```

Comment:
This should just return `DataTransactionBuilder.failNoData()` because we can't *remove* the flying state of an entity.